File: week3/Python/922. Sort Array By Parity II.py
class Solution:
    def sortArrayByParityII(self, a: List[int]) -> List[int]:
        j=1
        for i in range(0,len(a),2):
            if a[i] % 2:
                while a[j] % 2:
                    j+=2
                a[i],a[j]=a[j],a[i]
        return a

File: week3/Java/922. Sort Array By Parity II.java
class Solution {
    public int[] sortArrayByParityII(int[] A) {
        int even[]=new int[A.length/2];
        int odd[]=new int[A.length/2];
        int e=0,o=0;
        for(int i : A)
        {
            if(i%2==0)
            {
                even[e]=i;
                e++;
            }
            else
            {
                odd[o]=i;
                o++;
            }
        }
        e=0;
        o=0;
        for(int i=0;i<A.length;i++)
        {
            if(i%2==0)
            {
                A[i]=even[e];
                e++;
            }
            else
            {
                A[i]=odd[o];
                o++;
            }
        }
        return A;
    }
}
